### DISCUSSION TITLE
How can I use this integration with a Tasmota bulb that uses "mireds" instead of "kelvin"?

### DISCUSSION THREAD

**Discussion by robinogi** [basnijholt/adaptive-lighting#311] [General]
I've got a set of Tasmota bulbs that will accept a range of "mireds" from 153 (cool white) to 500 (warm white).  The Adaptive Lighting configuration asks for min and max kelvin, but it doesn't seem to convert automatically like I'd hoped it might.

How can I use the integration with these Tasmota bulbs?  Is there some way to submit the value as "mireds" instead of Kelvin?

A bit more challenging is that min mireds = max kelvin = coolest temp and max mireds = min kelvin = warmest temp (so the value directions are reversed).

Is there a solution I'm missing?

**Comment by Joshfindit** [comment]
I came here for this since by light also uses mireds:

```
min_mireds: 200
max_mireds: 500
supported_color_modes: color_temp
color_mode: color_temp
brightness: 255
color_temp: 314
hs_color: 27.547, 52.114
rgb_color: 255, 183, 122
xy_color: 0.479, 0.38
supported_features: 40
```

Apparently there is a straightforward calculation for mireds to Kelvin and back, and so I think a fair compromise is to add mireds to this part of the UI:

![image](https://user-images.githubusercontent.com/1937155/196989233-82e0165c-54ce-4350-a7c7-5ad9d7512a96.png)

Maybe one of those handy JS controls where entering a value in one sets the other automatically

**Comment by th3w1zard1** [comment]
Home assistant is _supposed_ to convert all service data in the light.turn_on call to stuff your bulb can understand. That's why it's backwards compatible with rgb_color, xy_color, rgbw_color, color_temp_kelvin, and color_temp_mireds. Are you getting a specific issue when adaptive-lighting is trying to adapt your light? Please post your debug logs (see the readme for instructions)
